### ISSUE TITLE
[Request]: add `MESA_LOADER_DRIVER_OVERRIDE=zink` when launching a game in Gamescope with an Intel iGPU

### ISSUE THREAD

**Issue by allaeddineomc** [bottlesdevs/Bottles#1834]
### Tell us the problem or your need

gamescope is rendering broken graphics on intel IGPUs and adding the variable INTEL_DEBUG=norbc when running gamescope on intel fixes the rendering 

### Describe the solution you'd like

this is an example of how the command should look to make it render properly 
``` INTEL_DEBUG=norbc gamescope '/var/home/edge/Games/steam/Besiege/start.sh' 
```

**Comment by allaeddineomc** [comment]
another solution can be adding a field to gamescope settings where the user can add arguments

**Comment by allaeddineomc** [comment]
@mirkobrombin  , i'm trying to implement this but the app doesn't seem to be building with gnome-builder nor with the included appimage script , anything i can do ?

**Comment by mirkobrombin** [comment]
Check if documentation for manual building the flatpak https://docs.usebottles.com/getting-started/installation

**Comment by TheEvilSkeleton** [comment]
> INTEL_DEBUG=norbc

This seems like a debug flag. I don't want to resort to debug flags as a "solution" for stuff like this. Do you happen to know it's safe to use? Perhaps even cite from someone who is qualified in this domain?

**Comment by allaeddineomc** [comment]
> > INTEL_DEBUG=norbc
> 
> This seems like a debug flag. I don't want to resort to debug flags as a "solution" for stuff like this. Do you happen to know it's safe to use? Perhaps even cite from someone who is qualified in this domain?

i managed to add the flag using flat seal , it's a bit janky solution but it's much better than the default behavior , mango hud isn't functioning with gamescope turned on but it's a fair trade for me

**Comment by TheEvilSkeleton** [comment]
Based on https://github.com/Plagman/gamescope/issues/356#issuecomment-1195054736, you should use `noccs` instead of `norbc`. Can you use MangoHud with Gamescope when you set it to `noccs`?

**Comment by allaeddineomc** [comment]
sorry for taking too long to reply , i didn't got notified , using ```noccs``` renders broken graphics that look different form the broken graphics when ran with no environment variables , ```mango hud``` isn't working with ```noccs``` just like with ```norbc```

**Comment by allaeddineomc** [comment]
gamescope disabled 
![Screenshot from 2022-08-03 22-31-06](https://user-images.githubusercontent.com/57415185/182716730-269f54a1-4e23-4238-b33f-78798cc8d338.png)
gamescope enabled , no variables set 
![Screenshot from 2022-08-03 22-31-38](https://user-images.githubusercontent.com/57415185/182716792-09c51f76-e43b-4e57-8ba4-95b14fc7220f.png)
gamescope enabled , INTEL_DEBUG=norbc
![Screenshot from 2022-08-03 22-33-58](https://user-images.githubusercontent.com/57415185/182716873-d8dc701b-9d6d-4f61-b2d4-57943c57c7f7.png)
gamescope enabled , INTEL_DEBUG=noccs
![Screenshot from 2022-08-03 22-35-18](https://user-images.githubusercontent.com/57415185/182716965-a5834c51-7dc9-4c2e-bc2e-bbc037db0f6c.png)
are there any configs or logs i need to send to help with debugging ?

**Comment by TheEvilSkeleton** [comment]
What happens when you run with `MESA_LOADER_DRIVER_OVERRIDE=zink` and `INTEL_DEBUG=noccs`?

**Comment by allaeddineomc** [comment]
> What happens when you run with `MESA_LOADER_DRIVER_OVERRIDE=zink` and `INTEL_DEBUG=noccs`?

it renders correctly but no mangohud , using ```MESA_LOADER_DRIVER_OVERRIDE=zink``` without  ```INTEL_DEBUG=noccs``` also renders correctly with no mangohud

**Comment by xiota** [comment]
Maybe an option to run programs with arbitrary environment variables would be better than hard coding `noccs`/`norbc`.  Unfortunately `-e` has already been assigned to Steam (why not `-S`?).

1. It could be used for other workarounds that are not yet known.
2. It could be used to set game-specific environment variables.
3. Not all games require the `INTEL_DEBUG=noccs` workaround.
4. The `nocss` workaround does not work for all games.  DOSBox has corrupted graphics, even when `INTEL_DEBUG=noccs` is set.  Even the command prompt colors are wrong, but it's easier to tell when running a game, like [Ms Pac PC](https://jrok.com/games_orig.html).  DOSBox seems to work correctly with Zink.
5. The Zink workaround (at least on my computer) needs to be set for the game environment, but not gamescope, like so:

       gamescope -U -- sh -c "MESA_LOADER_DRIVER_OVERRIDE=zink dosbox -exit game.exe"

   It does not work when run like this:

       MESA_LOADER_DRIVER_OVERRIDE=zink gamescope -- dosbox -exit game.exe
6. In addition to setting variables normally with "VAR=value", maybe common workarounds could have short names that are easier to remember and type, like "intel-noccs" and "mesa-zink".

**Comment by xiota** [comment]
@allaeddineomc From mesa docs, `noccs` and `norbc` should do the same thing.  You may be using an old version of mesa from before `noccs` was added.

**Comment by TheEvilSkeleton** [comment]
@xiota thanks for the info. Where did you find it in the Mesa docs? I'm looking at https://mesa.readthedocs.io but it's infinitely buffering.

@allaeddineomc where did you get Bottles from?

**Comment by allaeddineomc** [comment]
> @allaeddineomc where did you get Bottles from?
i'm always running the latest flatpak from flathub

**Comment by allaeddineomc** [comment]
> @allaeddineomc From mesa docs, `noccs` and `norbc` should do the same thing. You may be using an old version of mesa from before `noccs` was added.

on my system (fedora silverblue) mesa is 22.1.3

**Comment by TheEvilSkeleton** [comment]
> > @allaeddineomc From mesa docs, `noccs` and `norbc` should do the same thing. You may be using an old version of mesa from before `noccs` was added.
> 
> on my system (fedora silverblue) mesa is 22.1.3

Flathub packages Mesa 21. Let's wait for 22 to come out on Flathub, and then we can make some workarounds and whatnot.

**Comment by xiota** [comment]
Whatever workaround that's ultimately implemented should be flexible.  Some games work with `INTEL_DEBUG=noccs` (or `norbc`), some work with `MESA_LOADER_DRIVER_OVERRIDE=zink`, some work with either, some work with *neither*.  I've also found that Zink has to be run in a subshell:

    gamescope -U -- sh -c 'MESA_LOADER_DRIVER_OVERRIDE=zink game_exec'

**Comment by TheEvilSkeleton** [comment]
Good point. As for MangoHud, it depends on `mangoapp` https://github.com/bottlesdevs/Bottles/issues/1437.

**Comment by TheEvilSkeleton** [comment]
> gamescope enabled , INTEL_DEBUG=norbc
> ![Screenshot from 2022-08-03 22-33-58](https://user-images.githubusercontent.com/57415185/182716873-d8dc701b-9d6d-4f61-b2d4-57943c57c7f7.png)

@allaeddineomc can you tell me how you set this variable?

**Comment by allaeddineomc** [comment]
sorry for being too late , i forgot about the issue , i used flatseal environment variables 
![Screenshot from 2022-08-27 14-13-12](https://user-images.githubusercontent.com/57415185/187031840-b4394614-f775-47fe-aacc-317f7b695af7.png)

**Comment by mirkobrombin** [comment]
I think this is something at user end
